### ISSUE TITLE
Failed to get function name while analysing file

### ISSUE THREAD

**Issue by swissspidy** [johnbillion/wp-compat#7]
Just briefly tried to install this at https://github.com/GoogleForCreators/web-stories-wp but getting the following error:

```
     Internal error: Failed to get method name while analysing file /Users/pascalb/Workspace/WordPress/web-stories-wp/includes/Stories_Script_Data.php  
                                                                                                                                                        
     Run PHPStan with -v option and post the stack trace to:                                                                                            
     https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml                                                                             
                                                                                                                                                        
     Internal error: Failed to get function name while analysing file                                                                                   
     /Users/pascalb/Workspace/WordPress/web-stories-wp/includes/Infrastructure/Injector/SimpleInjector.php                                              
                                                                                                                                                        
     Run PHPStan with -v option and post the stack trace to:                                                                                            
     https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml 
```

Beginning of the stack trace is as follows:

```
   Internal error: Failed to get function name while analysing file                                                                                             
     /Users/pascalb/Workspace/WordPress/web-stories-wp/includes/Infrastructure/Injector/SimpleInjector.php                                                        
     Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                                    
     ## /Users/pascalb/Workspace/WordPress/web-stories-wp/vendor/johnbillion/wp-compat/src/Rules/SinceVersionRule.php(179)                                        
     #0 /Users/pascalb/Workspace/WordPress/web-stories-wp/vendor/johnbillion/wp-compat/src/Rules/SinceVersionRule.php(75):                                        
     WPCompat\PHPStan\Rules\SinceVersionRule::getFunctionName(Object(PhpParser\Node\Expr\FuncCall))                                                               
     #1 /Users/pascalb/Workspace/WordPress/web-stories-wp/vendor/johnbillion/wp-compat/src/Rules/SinceVersionRule.php(60):                                        
     WPCompat\PHPStan\Rules\SinceVersionRule->processFuncCall(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                       
     #2 phar:///Users/pascalb/Workspace/WordPress/web-stories-wp/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(120):                          
     WPCompat\PHPStan\Rules\SinceVersionRule->processNode(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                           
     #3 phar:///Users/pascalb/Workspace/WordPress/web-stories-wp/vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassStatementsGatherer.php(119):                   
     PHPStan\Analyser\FileAnalyser->PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                      
 ```

So the exception is happening here:

https://github.com/johnbillion/wp-compat/blob/f290c54d67a549376a57a0870ea7a75f6a0001ea/src/Rules/SinceVersionRule.php#L171-L180

Current version of the two files in question:

https://github.com/GoogleForCreators/web-stories-wp/blob/c59bd62e1a7a98c0bf8001b968d51a9083c610aa/includes/Stories_Script_Data.php

https://github.com/GoogleForCreators/web-stories-wp/blob/c59bd62e1a7a98c0bf8001b968d51a9083c610aa/includes/Infrastructure/Injector/SimpleInjector.php

**Comment by johnbillion** [comment]
Thanks for the report. It's being tripped up by [this variable function name syntax](https://github.com/GoogleForCreators/web-stories-wp/blob/c59bd62e1a7a98c0bf8001b968d51a9083c610aa/includes/Stories_Script_Data.php#L120-L122). I'll take a look at a fix for it.

**Comment by johnbillion** [comment]
Fixed in 42a5ceb. I ran 0.2.0 over web-stories-wp and it's only showing those variable function name calls as errors now (instead of throwing an exception). I'll look into whether PHPStan identify them specifically so they can be ignored.
